### PR TITLE
JDK-8284112 Minor cleanup could be done in javax.crypto

### PR DESCRIPTION
**Pull request by mcpowers**
---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8284112](https://bugs.openjdk.java.net/browse/JDK-8284112): Minor cleanup could be done in javax.crypto

### Reviewers
 * [Bradford Wetmore](https://openjdk.java.net/census#wetmore) (@bradfordwetmore - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/8214/head:pull/8214` \
`$ git checkout pull/8214`

Update a local copy of the PR: \
`$ git checkout pull/8214` \
`$ git pull https://git.openjdk.java.net/jdk pull/8214/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8214`

View PR using the GUI difftool: \
`$ git pr show -t 8214`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/8214.diff">https://git.openjdk.java.net/jdk/pull/8214.diff</a>

</details>

### CODE REVIEW

File: src/java.base/share/classes/javax/crypto/CryptoPermission.java [hunk 1]
```diff
@@ -442,20 +433,13 @@ private boolean impliesParameterSpec(boolean checkParam,
...
-                return true;
-            }
-            return false;
-        } else if (this.checkParam) {
-            return false;
-        } else {
-            return true;
-        }
+            return this.algParamSpec.equals(algParamSpec);
+        } else return !this.checkParam;
```
Comment (bradfordwetmore) [inline src/java.base/share/classes/javax/crypto/CryptoPermission.java:437]: Please use the standard coding format.

`
} else { 
    return !this.checkParam;
}
`

File: src/java.base/share/classes/javax/crypto/CryptoPermissions.java [hunk 2]
```diff
@@ -156,7 +155,7 @@ public void add(Permission permission) {
     }
 
     /**
-     * Checks if this object's PermissionCollection for permissons
+     * Checks if this object's PermissionCollection for permissions
```
Comment (bradfordwetmore) [inline src/java.base/share/classes/javax/crypto/CryptoPermissions.java:158]: Just FYI and not for this review, but this class should really be updated to use proper javadoc comment style, which is to tag all objects with <code>PermissionCollection</code>

**Comment by bradfordwetmore** [comment]
Can you file a bug to update the javax.crypto files to use proper javadoc for mentioned classes, e.g. <code> tags.

File: src/java.base/share/classes/javax/crypto/ExemptionMechanism.java [hunk 3]
```diff
@@ -139,7 +139,7 @@ public final String getName() {
      *
      * @see java.security.Provider
      */
-    public static final ExemptionMechanism getInstance(String algorithm)
+    public static ExemptionMechanism getInstance(String algorithm)
```
Comment (bradfordwetmore) [inline src/java.base/share/classes/javax/crypto/ExemptionMechanism.java:142]: Others might disagree with this comment, but I would encourage you not to make these changes throughout your PR.  (@jddarcy ?) 

Even though a static method is implicitly final and IJ is correct, the final keyword does prevent subclasses from inadvertently using the same method signature.

File: src/java.base/share/classes/javax/crypto/SealedObject.java [hunk 4]
```diff
@@ -442,13 +440,13 @@ private ObjectInputStream getExtObjectInputStream(Cipher c)
...
     static {
-        SharedSecrets.setJavaxCryptoSealedObjectAccess((obj,c) -> obj.getExtObjectInputStream(c));
+        SharedSecrets.setJavaxCryptoSealedObjectAccess(SealedObject::getExtObjectInputStream);
     }
 }
 
 final class extObjectInputStream extends ObjectInputStream {
     extObjectInputStream(InputStream in)
-        throws IOException, StreamCorruptedException {
+        throws IOException {
```
Comment (bradfordwetmore) [inline src/java.base/share/classes/javax/crypto/SealedObject.java:449]: This is another "I probably wouldn't do that..."  

It's nice to know what kind of IOExceptions you might get, so leaving StreamCorruptedException is ok.

File: src/java.base/share/classes/javax/crypto/spec/DESKeySpec.java [hunk 5]
```diff
@@ -41,12 +41,12 @@ public class DESKeySpec implements java.security.spec.KeySpec {
...
     public static final int DES_KEY_LEN = 8;
 
-    private byte[] key;
+    private final byte[] key;
 
     /*
      * Weak/semi-weak keys copied from FIPS 74.
      *
-     * "...The first 6 keys have duals different than themselves, hence
+     * "...The first 6 keys have duals different from themselves, hence
```
Comment (bradfordwetmore) [inline src/java.base/share/classes/javax/crypto/spec/DESKeySpec.java:49]: Suggest you leave this alone as this is a direct quote from FIPS 74 (page 10).

File: src/java.base/share/classes/javax/crypto/CipherInputStream.java [hunk 6]
```diff
@@ -156,8 +154,6 @@ private int getMoreData() throws IOException {
         ensureCapacity(readin);
         try {
             ofinish = cipher.update(ibuffer, 0, readin, obuffer, ostart);
-        } catch (IllegalStateException e) {
```
Comment (bradfordwetmore) [inline src/java.base/share/classes/javax/crypto/CipherInputStream.java:159]: This is another one of those I would probably leave alone, just so it's clear what should be done.

File: src/java.base/share/classes/javax/crypto/CipherSpi.java [hunk 7]
```diff
@@ -25,16 +25,9 @@
...
-import java.security.Key;
-import java.security.SecureRandom;
-import java.security.NoSuchAlgorithmException;
-import java.security.InvalidKeyException;
-import java.security.InvalidAlgorithmParameterException;
-import java.security.ProviderException;
-import java.security.spec.AlgorithmParameterSpec;
-
 import java.nio.ByteBuffer;
+import java.security.*;
```
Comment (bradfordwetmore) [inline src/java.base/share/classes/javax/crypto/CipherSpi.java:29]: This is another one that some people will complain about.  I personally prefer this style, others prefer everything written out as long as it's not "too many."

File: src/java.base/share/classes/javax/crypto/CryptoPermission.java (hunk 1 above)
Comment (mcpowers) [reply src/java.base/share/classes/javax/crypto/CryptoPermission.java:437]: Using the first format.

File: src/java.base/share/classes/javax/crypto/CryptoPermissions.java (hunk 2 above)
Comment (mcpowers) [reply src/java.base/share/classes/javax/crypto/CryptoPermissions.java:158]: No change required.

File: src/java.base/share/classes/javax/crypto/ExemptionMechanism.java (hunk 3 above)
Comment (mcpowers) [reply src/java.base/share/classes/javax/crypto/ExemptionMechanism.java:142]: Keeping the original here and elsewhere.

File: src/java.base/share/classes/javax/crypto/SealedObject.java (hunk 4 above)
Comment (mcpowers) [reply src/java.base/share/classes/javax/crypto/SealedObject.java:449]: Keeping the original. Still, why would IJ suggest removing this unless it had already figured out the exception could never happen?

File: src/java.base/share/classes/javax/crypto/spec/DESKeySpec.java (hunk 5 above)
Comment (mcpowers) [reply src/java.base/share/classes/javax/crypto/spec/DESKeySpec.java:49]: Keeping the original.

File: src/java.base/share/classes/javax/crypto/CipherInputStream.java (hunk 6 above)
Comment (mcpowers) [reply src/java.base/share/classes/javax/crypto/CipherInputStream.java:159]: Keeping the original.

File: src/java.base/share/classes/javax/crypto/CipherSpi.java (hunk 7 above)
Comment (mcpowers) [reply src/java.base/share/classes/javax/crypto/CipherSpi.java:29]: Keeping the change for now.

File: src/java.base/share/classes/javax/crypto/CryptoPolicyParser.java [hunk 8]
```diff
@@ -280,11 +279,11 @@ private CryptoPermissionEntry parsePermissionEntry(
         return e;
     }
 
-    private static final AlgorithmParameterSpec getInstance(String type,
-                                                            Integer[] params)
+    private static AlgorithmParameterSpec getInstance(String type,
```
Comment (bradfordwetmore) [inline src/java.base/share/classes/javax/crypto/CryptoPolicyParser.java:282]: Why are you removing final here?

**Review by bradfordwetmore — APPROVED** [review]
Except for the one issue, this looks ok to me.  Go ahead and push when you've resolved

**Pull request closed without merging**